content/v1.14/cli/command-reference.md

```diff
@@ -0,0 +1,416 @@
+### xpkg logout
+
+Use `crossplane xpkg logout` to invalidate the current `crossplane xpkg login` 
+session.
+
+{{< hint "note" >}}
+Using `crossplane xpkg logout` removes the `session` from the 
+`~/.crossplane/config.json` file, but doesn't delete the configuration file.
+{{< /hint >}}
+
+### xpkg push
+
+Push a Crossplane package file to a package registry. 
+
+The Crossplane CLI pushes images to the 
+[Upbound Marketplace](https://marketplace.upbound.io/) at `xpkg.upbound.io` by 
+default.
+
+{{< hint "note" >}}
+Pushing a package may require authentication with 
+[`crossplane xpkg login`](#xpkg-login)
+{{< /hint >}}
+
+Specify the organization, package name and tag with  
+`crossplane xpkg push <package>`
+
+By default the command looks in the current directory for a single `.xpkg` file 
+to push. 
+
+To push multiple files or to specify a specific `.xpkg` file use the `-f` flag.
+
+For example, to push a local package named `my-package` to 
+`crossplane-docs/my-package:v0.14.0` use:
+
+`crossplane xpkg push -f my-package.xpkg crossplane-docs/my-package:v0.14.0`
+
+#### Flags
+
+{{< table "table table-sm table-striped">}}
+| Short flag   | Long flag              | Description                                   |
+| ------------ | -------------          | ------------------------------                |
+| `-f`         | `--package-files=PATH` | A comma-separated list of xpkg files to push. |
+{{< /table >}}
+
+For example, to push a package named `my-configuration` use  
+`crossplane xpkg push -f my-configuration.xpkg`
+
+### xpkg update
+
+The `crossplane xpkg update` command downloads and updates an existing package.
+
+By default the `crossplane xpkg update` command uses the Kubernetes 
+configuration defined in `~/.kube/config`.  
+
+Define a custom Kubernetes configuration file location with the environmental 
+variable `KUBECONFIG`.
+
+Specify the package kind, package file and optionally the name of the package 
+already installed in Crossplane.
+
+`crossplane xpkg update <package-kind> <package-file> [<optional-name>`
+
+The package file must be an organization, image and tag on the 
+[Upbound Marketplace](https://marketplace.upbound.io/).
+
+For example, to update to version 0.42.0 of the 
+[AWS S3 provider](https://marketplace.upbound.io/providers/upbound/provider-aws-s3/v0.42.0):
+
+`crossplane xpkg update provider xpkg.upbound.io/upbound/provider-aws-s3:v0.42.0`
+
+
+## beta
+
+Crossplane `beta` commands are experimental. These commands may change the 
+flags, options or outputs in future releases. 
+
+Crossplane maintainers may promote or remove commands under `beta` in future 
+releases.
+
+### beta render 
+
+The `crossplane beta render` command previews the output of a 
+[composite resource]({{<ref "../concepts/composite-resources">}}) after applying 
+any 
+[composition patches]({{<ref "../concepts/compositions#changing-resource-fields">}}) 
+and [composition function]({{<ref "../concepts/composition-functions">}}) 
+changes.
```

Comment:
It doesn't apply patches. It completely ignores the `resources` field. `crossplane beta render` only works with composition functions - i.e. a Composition with `mode: Pipeline`.

Reply:
Updated:

> The `crossplane beta render` command previews the output of a 
> [composite resource]({{<ref "../concepts/composite-resources">}}) after applying 
> any [composition functions]({{<ref "../concepts/composition-functions">}}).
> 
> {{< hint "important" >}}
> The `crossplane beta render` command doesn't apply 
> [composition patches]({{<ref "../concepts/compositions#changing-resource-fields">}}.
> {{< /hint >}}